internal/backends/dotnet/project_files.go

```diff
@@ -0,0 +1,133 @@
+}
+
+// .csproj file structure
+type project struct {
+	XMLName  xml.Name           `xml:"Project"`
+	Packages []packageReference `xml:"ItemGroup>PackageReference"`
+}
+
+// looks for the .csproj file in the current directory
+func findSpecFile() string {
+	files, err := ioutil.ReadDir("./")
+	if err != nil {
+		util.Die("Can't read current directory")
+	}
+
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), ".csproj") {
+			return f.Name()
+		}
+	}
+
+	return ".csproj"
+}
+
+// loads the details of the project spec file
+func listSpecfile() map[api.PkgName]api.PkgSpec {
+	var pkgs map[api.PkgName]api.PkgSpec
+	projectFile := findSpecFile()
+	if util.Exists(projectFile) {
+
+		specReader, err := os.Open(projectFile)
+		if err != nil {
+			util.Die("Could not open %s, with error: %q", projectFile, err)
+		}
+		defer specReader.Close()
+
+		pkgs, err = ReadSpec(specReader)
+		if err != nil {
+			util.Die("Failed to read spec file %s, with error: %q", projectFile, err)
+		}
+	}
+	return pkgs
+}
+
+// reads the spec and builds up packages
+func ReadSpec(specReader io.Reader) (map[api.PkgName]api.PkgSpec, error) {
+	xmlbytes, err := ioutil.ReadAll(specReader)
+	if err != nil {
+		return nil, err
+	}
+
+	var project project
+	err = xml.Unmarshal(xmlbytes, &project)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to unmarshal file content %q", err)
+	}
+
+	pkgs := map[api.PkgName]api.PkgSpec{}
+	for _, packageReference := range project.Packages {
+		pkgName := packageReference.Include
+		pkgVersion := api.PkgVersion(packageReference.Version)
+		pkgs[api.PkgName(pkgName)] = api.PkgSpec(pkgVersion)
+	}
+	return pkgs, nil
+}
+
+// loads the details of the lock file
+func listLockfile() map[api.PkgName]api.PkgVersion {
+	pkgs := map[api.PkgName]api.PkgVersion{}
+	if util.Exists(lockFile) {
+		specReader, err := os.Open(lockFile)
+		if err != nil {
+			util.Die("Could not open %s, with error: %q", lockFile, err)
+		}
+		defer specReader.Close()
+
+		pkgs, err = ReadLock(specReader)
+		if err != nil {
+			util.Die("Error reading lockFile %s %q", lockFile, err)
+		}
+	}
+	return pkgs
+}
+
+// reads the lock file and buils up packages
+func ReadLock(lockFileReader io.Reader) (map[api.PkgName]api.PkgVersion, error) {
+	jsonBytes, err := ioutil.ReadAll(lockFileReader)
+	if err != nil {
+		return nil, err
+	}
+	var rawJson interface{}
+	err = json.Unmarshal(jsonBytes, &rawJson)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to unmarshal lock file data %q", err)
+	}
+
+	m := rawJson.(map[string]interface{})
+	dependencies := m["dependencies"].(map[string]interface{})
+	pkgs := map[api.PkgName]api.PkgVersion{}
+	for _, v := range dependencies {
+		packages := v.(map[string]interface{})
+		for packageName, details := range packages {
+			version := details.(map[string]interface{})["resolved"]
+			if version != nil {
+				pkgs[api.PkgName(packageName)] = api.PkgVersion(details.(map[string]interface{})["resolved"].(string))
+			}
+		}
+	}
```

Comment:
Why... is this using `interface{}` instead of a json struct?

Reply:
The dependencies section of the lock file has an intersting structure where the keys in the json file don't represent field names but are values instead.  Most of the structure of the file is important to the implementation.  I chose to implement this change by extracting just the data that was necessary to implement the functionality.  The lock file represents a map rather than a type safe datastructure.

Reply:
I has some time to spend on this today and changed the implementation to use a structure.  Take look.